Modules/Components/Niftyreg/include/selxNiftyregAladinComponent.hxx

```diff
@@ -0,0 +1,147 @@
+/*=========================================================================
+ *
+ *  Copyright Leiden University Medical Center, Erasmus University Medical
+ *  Center and contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "selxNiftyregAladinComponent.h"
+#include "selxCheckTemplateProperties.h"
+
+namespace selx
+{
+template< class TPixel >
+NiftyregAladinComponent< TPixel >::NiftyregAladinComponent( const std::string & name, LoggerImpl & logger ) : Superclass( name, logger )
+{
+  m_reg_aladin = new reg_aladin< TPixel >();
+}
+
+
+template< class TPixel >
+NiftyregAladinComponent< TPixel >::~NiftyregAladinComponent()
+{
+  delete m_reg_aladin;
+}
+
+
+template< class TPixel >
+int
+NiftyregAladinComponent< TPixel >
+::Set( typename NiftyregReferenceImageInterface< TPixel >::Pointer component )
+{
+  // store the shared_ptr to the data, otherwise it gets freed
+  this->m_reference_image = component->GetReferenceNiftiImage();
+  // connect the itk pipeline
+  this->m_reg_aladin->SetInputReference( this->m_reference_image.get() );
+  return 0;
+}
+
+
+template< class TPixel >
+int
+NiftyregAladinComponent< TPixel >
+::Set( typename NiftyregFloatingImageInterface< TPixel >::Pointer component )
+{
+  // store the shared_ptr to the data, otherwise it gets freed
+  this->m_floating_image = component->GetFloatingNiftiImage();
+  // connect the itk pipeline
+  this->m_reg_aladin->SetInputFloating(this->m_floating_image.get());
+  return 0;
+}
+
+
+template< class TPixel >
+std::shared_ptr< nifti_image >
+NiftyregAladinComponent< TPixel >
+::GetWarpedNiftiImage()
+{
+  return this->m_warped_image;
+}
+
+template< class TPixel >
+mat44 *
+NiftyregAladinComponent< TPixel >
+::GetAffineNiftiMatrix()
+{
+  return this->m_reg_aladin->GetTransformationMatrix();
+}
+
+template< class TPixel >
+void
+NiftyregAladinComponent<  TPixel >
+::RunRegistration()
```

Comment:
You could consider logging some statements in this function to test out the new logging functionality.

Reply:
done